.github/workflows/chain-spec-snapshot-build.yml

```diff
@@ -13,7 +13,9 @@ on:
 
 jobs:
   chains-spec:
-    runs-on: ${{ fromJson(github.repository_owner == 'autonomys' && '["self-hosted", "ubuntu-22.04-x86-64"]' || 'ubuntu-22.04') }}
+    runs-on: ${{ fromJson(github.repository_owner == 'autonomys' &&
```

Comment:
This is really ugly, is it possible to add those runners to the same tag as we used before to avoid this mess and hardcoded specs?

Reply:
The normal format would be:

>     runs-on:
>      - runs-on
>      - runner=4cpu-linux-x64
>      - run-id=${{ github.run_id }}
>      - family=c7 (optional)
      
1- We use the json format to check the repo owner, and fallback to GH runners
2- Our runners need to specify specs since we need certain machines with enough CPU and RAM to run the workflows, otherwise we could get any spot instance available from different instance families.

https://runs-on.com/configuration/job-labels/#how-it-works

Reply:
Next section after that literally explains how to avoid all that boilerplate and links to dedicated page with repository configuration. I don't like that we have to define specs in the repo at all, but in case we do, we should do it in a single place and not in every single job.

Reply:
Yes, was trying something similar with windows custom image:
https://github.com/autonomys/subspace/blob/muhammad/ci-test/.github/runs-on.yml

I will see if i can do it with the labels 👍🏽

Reply:
They say the file has to be in the default branch for this to work though